dali-mqtt-daemon.py

```diff
@@ -172,22 +188,44 @@ def main_loop(driver, max_lamps, mqtt_server, mqtt_port, ha_prefix):
               }
 
     try:
-        with open(args.config, 'r') as stream:
-            logger.debug("Loading configuration from <%s>", args.config)
-            config = yaml.load(stream)
-
-        MQTT_BASE_TOPIC = config["mqtt_base_topic"]
-
-        main_loop(config["dali_driver"], config["dali_lamps"], config["mqtt_server"], config["mqtt_port"], config["ha_discover_prefix"])
+        config = load_config_file(args.config)
+
+        driver_object = None
+        driver = args.dali_driver
+        logger.debug("Using <%s> driver", driver)
+        if driver == HASSEB:
+            from dali.driver.hasseb import SyncHassebDALIUSBDriver 
+            driver_object = SyncHassebDALIUSBDriver()
+        elif driver == TRIDONIC:
+            from dali.driver.tridonic import SyncTridonicDALIUSBDriver
+            driver_object = SyncTridonicDALIUSBDriver()
+        elif driver == DALI_SERVER:
+            from dali.driver.daliserver import DaliServer
+            driver_object = DaliServer("localhost", 55825)    
+
+        mqqtc = create_mqtt_client(driver_object, config["dali_lamps"], config["mqtt_server"], config["mqtt_port"], config["mqtt_base_topic"], config["ha_discover_prefix"])
+        mqqtc.loop_start()
+
+        watchdog_observer = Observer()
+        watchdog_event_handler = ConfigFileSystemEventHandler(mqqtc, config, args.config, driver_object, loop)
+        watchdog_observer.schedule(watchdog_event_handler, args.config)
+        watchdog_observer.start()
     except FileNotFoundError as e:
-        logger.info("Configuration file %s created, please reload daemon", args.config)
-    except KeyError as e:
-        missing_key = e.args[0]
-        config[missing_key] = args.__dict__[missing_key]
-        logger.info("Configuration file updated, please reload daemon")
-    finally:
         try:
             with io.open(args.config, 'w', encoding="utf8") as outfile:
                 yaml.dump(config, outfile, default_flow_style=False, allow_unicode=True)
+                logger.info("Configuration file %s created, please reload daemon", args.config)
         except Exception as err:
-            logger.error(f"Could not save configuration: {err}")
+            logger.error(f"Could not save configuration: {err}")   
+    while True:
+        continue
+
+
+if __name__ == "__main__":
+    loop = asyncio.get_event_loop()
+    executor = concurrent.futures.ThreadPoolExecutor(os.cpu_count() * 5)
```

Comment:
Don't think we need these many Executers for dali2mqtt :) even you add the file system watchdog, it's just 2 threads...

Reply:
I added variable for it and set default value to 2

Reply:
There is no need for a ThreadPool.... we have MQTT (which runs it's own managed thread) and the watchdog recently added

Reply:
Ok sorry, you are right